### PR TITLE
View recipe

### LINKED ISSUE #7: As a user, I want to be able to view a list of all the recipes I have recorded.
R for recipe class.

### LINKED ISSUE #45: Refactor and rename containsMultipleWordsIgnoreCase in StringUtil.java
I don't want to create a new PR for this, maybe someone could make this change in their PR?

"Would be better to name it as "exactlyMultipleWordsIgnoreCase"? "contains" is a little misleading." - Rahul

### PR DESCRIPTION
**Pull request by khenghun**
Enable viewing of all items, all recipes, or a specific item

### CODE REVIEW

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 1]
```diff
@@ -140,4 +142,32 @@ public GuiSettings getGuiSettings() {
     public void setGuiSettings(GuiSettings guiSettings) {
         model.setGuiSettings(guiSettings);
     }
+
```
Comment (blackonyyx) [inline src/main/java/seedu/address/logic/LogicManager.java:145]: Perhaps some jdocs here would be good to ensure readability of code

File: src/main/java/seedu/address/logic/Logic.java [hunk 2]
```diff
@@ -79,4 +82,7 @@
      * Set the user prefs' GUI settings.
      */
     void setGuiSettings(GuiSettings guiSettings);
+
```
Comment (blackonyyx) [inline src/main/java/seedu/address/logic/Logic.java:85]: Jdocs

File: src/main/java/seedu/address/logic/commands/ListItemCommand.java [hunk 3]
```diff
@@ -1,25 +1,23 @@
...
+import seedu.address.ui.View;
 
 
 /**
- * Lists all items in the address book to the user.
+ * Lists all items in the inventory to the user.
  */
 public class ListItemCommand extends Command {
 
     public static final String COMMAND_WORD = "listi";
```
Comment (blackonyyx) [inline src/main/java/seedu/address/logic/commands/ListItemCommand.java:14]: Maybe should have a MESSAGE_USAGE string like if someone does sth like 
listi jkdnkjvsdsn 
or something ( unsure if will return error here)

File: src/main/java/seedu/address/model/item/DetailedItem.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import seedu.address.model.tag.Tag;
+import seedu.address.ui.View;
+
+/**
+ * Represents an Detailed Item which may be displayed.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class DetailedItem extends Item {
+    public DetailedItem(int id, String name, Quantity quantity, String description, Set<Integer> locationIds,
```
Comment (blackonyyx) [inline src/main/java/seedu/address/model/item/DetailedItem.java:17]: Constructs ...

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+        assertFalse(addAliceCommand.equals(null));
+
+        // different person -> returns false
+        assertFalse(addAliceCommand.equals(addBobCommand));
+    }
+
+    /**
+     * A default model stub that have all of the methods failing.
+     */
+    private class ModelStub implements Model {
```
Comment (aqcd) [inline src/test/java/seedu/address/logic/commands/AddCommandTest.java]: import from the file with existing modelstub

File: src/main/java/seedu/address/model/item/Item.java [hunk 6]
```diff
@@ -9,13 +9,15 @@
...
+import seedu.address.model.Inventory;
 import seedu.address.model.tag.Tag;
+import seedu.address.ui.View;
 
 /**
  * Represents an Item in the Inventoryinator.
  * Guarantees: details are present and not null, field values are validated, immutable.
  */
-public class Item {
+public class Item extends Inventory {
```
Comment (blackonyyx) [inline src/main/java/seedu/address/model/item/Item.java:20]: This doesnt seem to follow OOP design, is there perhaps a better representation?

File: src/main/java/seedu/address/model/Inventory.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package seedu.address.model;
+
+import seedu.address.ui.View;
+
+/**
+ * Encapsulates a possible Inventory entry, either an Item or a Recipe.
+ */
+public abstract class Inventory {
```
Comment (aqcd) [inline src/main/java/seedu/address/model/Inventory.java]: deceptive naming
try to remove if not too necessary, is an unusual class to have

File: src/main/java/seedu/address/model/item/Item.java (hunk 6 above)
Comment (blackonyyx) [inline src/main/java/seedu/address/model/item/Item.java]: May not be in line with good OOP practices. Consider alternative representation or if it is not the proper namespace usage, amend

File: src/test/java/seedu/address/logic/commands/ViewDetailsCommandTest.java [hunk 8]
```diff
@@ -58,7 +59,7 @@ public void execute_filterItem_success() {
         Item validItem = new ItemBuilder().build();
         ModelStub modelStub = new ModelStubWithItem(validItem);
```
Comment (blackonyyx) [inline src/test/java/seedu/address/logic/commands/ViewDetailsCommandTest.java:60]: Bit more spacing would be good to have and perhaps line commets

File: src/test/java/seedu/address/model/recipe/IngredientTest.java [hunk 9]
```diff
@@ -27,9 +36,28 @@ public void isItem() {
...
+        // Ensure that a known item id and equivalent int value returns true
         assertTrue(INGREDIENT_1.isItem(sameItemId));
-        // Known item id and non equivilent int value returns false.
+        // Known item id and non equivalent int value returns false.
         assertFalse(INGREDIENT_1.isItem(differentItemId));
     }
+
+    @Test
+    public void testToString_success() {
+        ArrayList<Item> itemArrayList = new ArrayList<>();
```
Comment (blackonyyx) [inline src/test/java/seedu/address/model/recipe/IngredientTest.java:50]: Same as abv

**Review by blackonyyx — CHANGES_REQUESTED** [review]
Some Namespace changes should be made especially item extends Inventory

File: src/main/java/seedu/address/logic/commands/CommandResult.java [hunk 10]
```diff
@@ -4,38 +4,51 @@
 
 import java.util.Objects;
 
+import seedu.address.ui.View;
```
Comment (Rahul0506) [inline src/main/java/seedu/address/logic/commands/CommandResult.java]: Can you not just import the View.InventoryType ?

File: src/main/java/seedu/address/logic/commands/ViewDetailsCommand.java [hunk 11]
```diff
@@ -5,6 +5,7 @@
 import seedu.address.commons.core.Messages;
 import seedu.address.model.Model;
 import seedu.address.model.item.NameIsExactlyPredicate;
+import seedu.address.ui.View;
```
Comment (Rahul0506) [inline src/main/java/seedu/address/logic/commands/ViewDetailsCommand.java]: same as before

File: src/main/java/seedu/address/model/recipe/Ingredient.java [hunk 12]
```diff
@@ -31,4 +38,18 @@ public boolean isItem(int itemId) {
...
         return String.valueOf(this.getKey());
     }
+
+    /**
+     * Displays the name of the ingredient and the quantity, based on the item list given.
+     * @param filteredItemList The current item list in the model.
+     * @return String representation of the ingredient ready for display.
+     */
+    public String toString(ObservableList<Item> filteredItemList) {
+        HashMap<Integer, String> hm = new HashMap<>();
```
Comment (Rahul0506) [inline src/main/java/seedu/address/model/recipe/Ingredient.java]: maybe use a more descriptive name

**Review by Rahul0506 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/ListItemCommand.java [hunk 13]
```diff
@@ -1,25 +1,29 @@
...
 
+    public static final String MESSAGE_SUCCESS = "Listed all items";
+    public static final String MESSAGE_NO_ITEMS = "You have no items in your inventory now, will you like to add some?";
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.updateFilteredItemList(PREDICATE_SHOW_ALL_ITEMS);
-        return new CommandResult(MESSAGE_SUCCESS);
+        if (model.getFilteredItemList().size() == 0) {
```
Comment (aqcd) [inline src/main/java/seedu/address/logic/commands/ListItemCommand.java]: use isEmpty

File: src/main/java/seedu/address/logic/commands/ListRecipeCommand.java [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+ */
+public class ListRecipeCommand extends Command {
+
+    public static final String COMMAND_WORD = "listr";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + " :lists all recipes";
+
+    public static final String MESSAGE_SUCCESS = "Listed all recipes";
+    public static final String MESSAGE_NO_RECIPES = "You have no recipes in your inventory now, "
+            + "will you like to add some?";
```
Comment (aqcd) [inline src/main/java/seedu/address/logic/commands/ListRecipeCommand.java]: Might be better to not have "will you like to add some?" if there's no direct button to do so

File: src/main/java/seedu/address/logic/commands/ListRecipeCommand.java [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        if (model.getFilteredRecipeList().size() == 0) {
```
Comment (aqcd) [inline src/main/java/seedu/address/logic/commands/ListRecipeCommand.java]: use isEmpty

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java [hunk 16]
```diff
@@ -0,0 +1,122 @@
+package seedu.address.logic.commands;
```
Comment (aqcd) [inline src/test/java/seedu/address/logic/commands/AddCommandTest.java]: should this file be added?

**Review by aqcd — APPROVED** [review]

**Pull request merged**